extension/persistence/eclipselink/src/main/java/io/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkMetaStoreSessionImpl.java

```diff
@@ -118,13 +110,55 @@ public PolarisEclipseLinkMetaStoreSessionImpl(
   }
 
   /** Load the persistence unit properties from a given configuration file */
-  private Map<String, String> loadProperties(String confFile, String persistenceUnitName) {
-    if (properties != null) {
-      return properties;
+  private EntityManagerFactory createEntityManagerFactory(
+      @NotNull RealmContext realmContext,
+      @Nullable String confFile,
+      @Nullable String persistenceUnitName) {
+    ClassLoader prevClassLoader = Thread.currentThread().getContextClassLoader();
+    try {
+      persistenceUnitName = persistenceUnitName == null ? "polaris" : persistenceUnitName;
+      confFile = confFile == null ? "META-INF/persistence.xml" : confFile;
+
+      // Currently eclipseLink can only support configuration as a resource inside a jar. To support
+      // external configuration, persistence.xml needs be placed inside a jar and here is to add the
+      // jar to the classpath.
+      // Supported configuration file: META-INFO/persistence.xml, /tmp/conf.jar!/persistence.xml
+      int splitPosition = confFile.indexOf("!/");
+      if (splitPosition != -1) {
+        String jarPrefixPath = confFile.substring(0, splitPosition);
+        confFile = confFile.substring(splitPosition + 2);
+        URL prefixUrl = this.getClass().getClassLoader().getResource(jarPrefixPath);
+        if (prefixUrl == null) {
+          prefixUrl = new File(jarPrefixPath).toURI().toURL();
+        }
+        URLClassLoader currentClassLoader =
+            new URLClassLoader(new URL[] {prefixUrl}, this.getClass().getClassLoader());
+        Thread.currentThread().setContextClassLoader(currentClassLoader);
```

Comment:
Let's add a logging statement here stating that we're changing the classloader temporarily